### PR TITLE
Code quality workflow

### PR DESCRIPTION
**Pull request by elisim**
Hi,
Thank you for the great template, I use it a lot 😁

I enhanced the `check code formatting` to run only on the changed files, and move it to separate a workflow file.
From experience, sometimes pre-commit's output is large, so IMHO the separation is needed :) 

I also fixed a small English mistake.

Thank you,
Eli

Edit: consider [badging the repository](https://pre-commit.com/#badging-your-repository) with pre-commit ;)

### CODE REVIEW

**Comment by ashleve** [comment]
@elisim Thank you for the PR!

I like moving pre-commit checks into separate file.

I'm not sure if running pre-commits only for modified files is the right choice for this template. 
1. From what I understand, this means CI won't detect the incorrectly formatted code after it's already committed to the main branch? It also means pre-commit CI works only through PRs (instead of also commits to the main branch), so this part of the config becomes redundant:
https://github.com/ashleve/lightning-hydra-template/blob/3e6f6bc8d3d0d60bce3a42af23bfc585da115d99/.github/workflows/code-quality.yaml#L10-L11
I'm not sure if this is a desired behavior, might be confusing for some users. 
2. Execution of currently configurated pre-commits usually takes up to a couple of seconds (I imagine it might be significantly longer for a large codebase though)

I think we can go with two different pre-commit workflows - the simple one for the main branch and the one running on changed files for PRs.

**Comment by elisim** [comment]
Thank you for your reply! 

> this part of the config becomes redundant

Yes. I assume that every new code to main is done via PRs. It's natural when working in a team, although when working solo, I agree - it might be a little confusing. 

> I think we can go with two different pre-commit workflows 

Sounds good. So [.github/workflows/code-quality.yaml](https://github.com/ashleve/lightning-hydra-template/pull/364/files#diff-4a2765c2cfcbd3804a66aab805cb92ddda74de1730923cc5bf53671d0beccf06) will remain, and reverting the changes in [.github/workflows/test.yml](https://github.com/ashleve/lightning-hydra-template/pull/364/files#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88)?

**Comment by ashleve** [comment]
> So [.github/workflows/code-quality.yaml](https://github.com/ashleve/lightning-hydra-template/pull/364/files#diff-4a2765c2cfcbd3804a66aab805cb92ddda74de1730923cc5bf53671d0beccf06) will remain, and reverting the changes in [.github/workflows/test.yml](https://github.com/ashleve/lightning-hydra-template/pull/364/files#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88)?

How about this:
- `test.yaml` (same as now, no reverting)
- `code-quality-main.yaml` (triggered on commit to main branch, runs on all files)
- `code-quality-pr.yaml` (triggered on pull request, runs on modified files)

?

**Comment by elisim** [comment]
Done

**Comment by ashleve** [comment]
Since pre-commits are now covered in separate workflows, we should remove them from `test.yaml`:
https://github.com/ashleve/lightning-hydra-template/blob/b0e7d4e8d02f0ce0f8f584da25a50880049fbae0/.github/workflows/test.yml#L71-L78

(sorry if I somehow confused you about this)

File: .github/workflows/code-quality-main.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Same as `code-quality-pr.yaml.yaml` but triggered on commit
```
Comment (ashleve) [inline .github/workflows/code-quality-main.yaml:1]:
```suggestion
# Same as `code-quality-pr.yaml` but triggered on commit
```

File: .github/workflows/code-quality-pr.yaml.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# This workflow finds which files was changed, print them, and then
+# run `pre-commit` on those files.
```
Comment (ashleve) [inline .github/workflows/code-quality-pr.yaml.yaml:2]:
```suggestion
# This workflow finds which files were changed, prints them, 
# and runs `pre-commit` on those files.
```

**Comment by ashleve** [comment]
Double yaml in filename: `code-quality-pr.yaml.yaml`

File: .github/workflows/code-quality-pr.yaml.yaml (hunk 2 above)
Comment (elisim) [reply .github/workflows/code-quality-pr.yaml.yaml]: done in https://github.com/ashleve/lightning-hydra-template/pull/364/commits/479def86850bfe89ca07d1e7a30eb83c5a5f640f

File: .github/workflows/code-quality-main.yaml (hunk 1 above)
Comment (elisim) [reply .github/workflows/code-quality-main.yaml]: done in https://github.com/ashleve/lightning-hydra-template/pull/364/commits/479def86850bfe89ca07d1e7a30eb83c5a5f640f

**Comment by elisim** [comment]
Thanks for the corrections :)

**Review by ashleve — APPROVED** [review]

**Comment by ashleve** [comment]
Merged. Thank you for the contribution :)

Also good idea with the pre-commit badge, I added it on this branch:
https://github.com/ashleve/lightning-hydra-template/tree/update-readme

**Pull request merged**